source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -0,0 +1,266 @@
+to specify which change events the change stream receives.
+
+To learn which aggregation operators your {+mdb-server+} version supports, see
+:ref:`<change-stream-modify-output>`.
+
+
+Example
+~~~~~~~
+
+The following code example shows how you can apply an aggregation pipeline to
+your change stream receive change events for insert and update operators
+only:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/change-streams/ChangeStreams.java
+   :language: java
+   :dedent:
+   :start-after: begin aggregationExample
+   :end-before: end aggregationExample
+
+When the change stream receives an update change event, the preceding code
+example outputs the following text:
+
+.. code-block:: text
+   :copyable: false
+
+   Received a change to the collection: ChangeStreamDocument{
+   operationType=update, resumeToken={...},
+   ...
+
+.. _java-change-stream-configure-pre-post:
+
+Include Pre-images and Post-images
+----------------------------------
+
+You can configure the change event to contain or omit the following data:
+
+- The **pre-image** which is a document that represents the version of the
+  document before the operation if it exists
+- The **post-image** which is a document that represents the version of the
+  document after the operation if it exists
+
+To receive change stream events that include a pre-image or post-image, you
+must connect to a MongoDB v6.0 or later deployment and set up the following:
+
+- Enable pre-images and post-images for the collection on your MongoDB
+  deployment.
+
+  .. see::
+
+     To learn how to enable these on your deployment, see the
+     :manual:`Change Streams with Document Pre- and Post-Images </changeStreams/#change-streams-with-document-pre--and-post-images>`
+     MongoDB server manual page.
+
+     To learn how to instruct the driver to create a collection with pre-images
+     and post-images enabled, see the :ref:`<java-change-stream-pre-post-collection>`
+     section.
+
+- Configure your change stream to retrieve either or both the pre-images and
+  post-images.
+
+  .. see::
+
+     To configure your change stream to include the pre-image, see
+     the :ref:`<java-pre-image-example>`.
+
+     To configure your change stream to include the post-image, see the
+     :ref:`<java-post-image-example>`.
+
+.. _java-change-stream-pre-post-collection:
+
+Create a Collection with Pre-Image and Post-Images Enabled
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create a collection with the pre-image and post-image option using the
+driver, specify and instance of  ``ChangeStreamPreAndPostImagesOptions``
+and call the ``createCollection()`` method as shown in the following example:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/change-streams/ChangeStreams.java
+   :language: java
+   :dedent:
+   :start-after: begin createCollection
+   :end-before: end createCollection
+
+.. warning::
```

Comment:
This warning seems out of place to me. The section is describing how to create a collection with pre and post-imaging, but this warning seems to just be about modifying a collection that has already been created. Does this warning only apply to collections with pre or post-imaging enabled? If so I think that could be clarified. 
If this is more of a general warning about modifying any collection with a change stream open, maybe it would make more sense in the "Open a Change Stream" section above?

Reply:
I moved some of the content around to separate the content that mentions how you can enable the on an existing collection and the warning about how, if configured to require the pre or post image, that they could fail in your application when making that change.

Reply:
I see, I think it makes more sense with these changes!